src/adapter.rs

```diff
@@ -10,6 +10,45 @@ use error::*;
 
 use bindings::*;
 
+/// Represent an operational status of the adapter
+/// See IP_ADAPTER_ADDRESSES docs for more details
+#[derive(Debug, Clone, Copy)]
+pub enum OperStatus {
+    IfOperStatusUp = 1,
+    IfOperStatusDown = 2,
+    IfOperStatusTesting = 3,
+    IfOperStatusUnknown = 4,
+    IfOperStatusDormant = 5,
+    IfOperStatusNotPresent = 6,
+    IfOperStatusLowerLayerDown = 7,
+}
+
+/// Represent an interface type
+/// See IANA docs on iftype for more details
+/// https://www.iana.org/assignments/ianaiftype-mib/ianaiftype-mib
+/// Note that we only support a subset of the IANA interface
+/// types and in case the adapter has an unsupported type,
+/// `IfType::Unsupported` is used. `IfType::Other`
+/// is different from `IfType::Unsupported`, as the former
+/// one is defined by the IANA itself.
+#[derive(Debug, Clone, Copy)]
+pub enum IfType {
+    Other = 1,
+    EthernetCsmacd = 6,
+    Iso88025Tokenring = 9,
+    Ppp = 23,
+    SoftwareLoopback = 24,
+    Atm = 37,
+    Ieee80211 = 71,
+    Tunnel = 131,
+    Ieee1394 = 144,
+    Unsupported,
```

Comment:
Hmm perhaps it should be replaced with `Unknown(u32)`?
That way it will contains the numeric value even if it's an uncommon interface type.
What do you think?